test/upgrade/simpleResources.yaml

```diff
@@ -0,0 +1,67 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: simple-task
+  namespace: upgrade
+spec:
+  params:
+    - name: rsp
+      type: string
+      default: "response"
+  steps:
+  - name: echo-param
+    image: alpine
+    script: |
+      echo "$(params.rsp)"
+  - name: check-workspace
+    image: alpine
+    script: |
+      if [ "$(workspaces.workspace.bound)" == "true" ]; then
+        echo "Workspace provided"
+      fi
+  sidecars:
+  - name: server
+    image: alpine:3.12.0
+    command: ['/bin/bash']
+    workingDir: /foo
+    script: echo server
+  workspaces:
+  - name: workspace
+    mountPath: /foo
+---
+apiVersion: tekton.dev/v1
+kind: Pipeline
+metadata:
+  name: simple-pipeline
+  namespace: upgrade
+spec:
+  description: foo
+  tasks:
+  - name: check-workspace
+    taskSpec:
+      params:
+        - name: task1-result
+          type: string
+          default: "1"
+      steps:
+        - image: alpine
+          name: check-workspace
+          script: |
+            if [ "$(workspaces.workspace.bound)" == "true" ]; then
+              echo "Workspace provided \n"
+              echo $(params.task1-result)
+            fi
+    timeout: 180s
```

Comment:
this isn't a supported field in v1, curious why it works?

Reply:
It's not a field of Pipeline either; we support v1 timeouts via `pipelinerun.spec.timeouts.pipeline`. In addition to fixing the syntax, can you figure out why tests passed when the incorrect syntax was used?

Reply:
Ahh sorry previously it should work as a field for the pipelineTask? https://github.com/tektoncd/pipeline/blob/d2cb90de128c85d1e3915fae21c0601c0a22845f/pkg/apis/pipeline/v1/pipeline_types.go#L208

I think for fields that does not exist like the current state, unknown fields are just dropped?

Reply:
It looks like you're trying to use this field in pipeline.spec, rather than pipeline.spec.tasks.

Unknown fields are not dropped when using kubectl create: 
```
$ k create -f tmp.yaml
Error from server (BadRequest): error when creating "tmp.yaml": admission webhook "webhook.pipeline.tekton.dev" denied the request: mutation failed: cannot decode incoming new object: json: unknown field "timeout"
```

Reply:
I restored it to the tasks.

I also added the fail test on creation failure.